CHANGELOG.md

```diff
@@ -14,15 +14,38 @@
 ## 🔦 Highlights <!-- omit in toc -->
 
 ### Smart Dialing <!-- omit in toc -->
-* When connecting to a peer we now do [happy eyeballs](https://www.rfc-editor.org/rfc/rfc8305) like dial prioritisation to prefer QUIC addresses over TCP addresses. We dial the QUIC address first and wait 250ms to dial the TCP address of the peer.
-* In our experiments we've seen little impact on latencies up to 80th percentile. 90th and 95th percentile latencies are impacted. For details see discussion on the [PR](https://github.com/libp2p/go-libp2p/pull/2260#issuecomment-1528848170).
-* For details of the address ranking logic see godoc for `swarm.DefaultDialRanker`.
-* To disable smart dialing and keep the old behaviour use the
-`libp2p.NoDelayNetworkDialRanker` option.
+
+This release introduces smart dialing logic. Currently, libp2p dials all addresses of a remote peer in parallel, and
+aborts all outstanding dials as soon as the first one succeeds.
+Dialing many addresses in parallel creates a lot of churn on the client side, and unnecessary load on the network and
+on the server side, and is heavily discouraged by the networking community (see [RFC 8305](https://www.rfc-editor.org/rfc/rfc8305) for example).
+
+When connecting to a peer we first determine the order to dial its addresses. This ranking logic considers a number of corner cases
+described in detail in the documentation of the swarm package (`swarm.DefaultDialRanker`).
+At a high level, this is what happens:
+* If a peer offers a WebTransport and a QUIC address (on the same IP:port), the QUIC address is preferred.
+* If a peer has a QUIC and a TCP address, the QUIC address is dialed first. Only if the connection attempt doesn't succeed within 250ms, a TCP connection is started.
+
+Our measurements on the IPFS network show that for >90% of established libp2p connections, the first connection attempt succeeds,
+leading a dramatic decrease in the number of aborted connection attempts.
```

Comment:
Can we put numbers to this to inform how dramatic this was?  For example, I think the "Cancel fraction" we had in https://github.com/libp2p/go-libp2p/pull/2260#issuecomment-1528848170 was useful.  

I'm thinking about what we'd say in a presentation or all hands.  Is this reducing the number of "wasted" connections by 50%, 75%, etc.

Reply:
I ran the numbers again for master branch. This is reducing the number of "wasted" connections by 50%.
With no smart dialing 60% of all connection attempts are cancelled. With smart dialing turned on 30% of all connection attempts are cancelled.

Reply:
We can report numbers here, or on the next release when we have black hole detection (or both). I expect the numbers to be significantly better when we have that, as there are quite a few dual stack hosts in the network.